File: app/src/main/java/com/shimhg02/solorestorant/utils/Service/FCMTokenService.java
package com.shimhg02.solorestorant.utils.Service;


import android.util.Log;
import com.google.firebase.iid.FirebaseInstanceId;
import com.google.firebase.messaging.FirebaseMessagingService;


public class FCMTokenService extends FirebaseMessagingService {

    private static final String TAG = "MyFirebaseIIDService";
    private static final String FRIENDLY_ENGAGE_TOPIC = "friendly_engage";

    /**
     * The Application's current Instance ID token is no longer valid and thus a new one must be requested.
     */
    public void onTokenRefresh() {
        String token = FirebaseInstanceId.getInstance().getToken();
        Log.d(TAG, "Token : "+token);
        //FirebaseMessaging.getInstance().subscribeToTopic(FRIENDLY_ENGAGE_TOPIC);

    }

}

File: app/src/main/java/com/shimhg02/solorestorant/utils/Service/FCMTokenService.kt
package com.shimhg02.solorestorant.utils.Service

import android.util.Log
import com.google.firebase.iid.FirebaseInstanceId
import com.google.firebase.messaging.FirebaseMessaging
import com.google.firebase.messaging.FirebaseMessagingService

@Suppress("DEPRECATION")
class FCMTokenService : FirebaseMessagingService() {

    fun onTokenRefresh() {
        val token = FirebaseInstanceId.getInstance().token
        Log.d(TAG, "Token Test : $token")
        FirebaseMessaging.getInstance().subscribeToTopic(FRIENDLY_ENGAGE_TOPIC);
    }

    companion object {
        private const val TAG = "MyFirebaseIIDService"
        const val FRIENDLY_ENGAGE_TOPIC = "friendly_engage"
    }
}